### PR TITLE
BZ-21767: feat: Add-support-for-animation-direction-in-toast-component

### PR DESCRIPTION
**Pull request by badri-singhal**
- Directions supported is left to right, right to left, top to bottom, bottom to top
- Toast position can be over content or inside content
- Dev Proof:- https://drive.google.com/file/d/1KjDnA8ywAX2kK2ZbvOq60xxgMMuv5MnN/view?usp=sharing

### CODE REVIEW

File: src/lib/Toast/Toast.svelte [hunk 1]
```diff
@@ -6,6 +6,22 @@
...
+    },
+    'right-to-left': { in: { x: 500, y: 0, duration: 400 }, out: { x: 500, y: 0, duration: 800 } },
+    'top-to-bottom': {
+      in: { x: 0, y: -500, duration: 400 },
+      out: { x: 0, y: -100, duration: 800 }
+    },
+    'bottom-to-top': { in: { x: 0, y: 500, duration: 400 }, out: { x: 0, y: 500, duration: 800 } }
+  };
+
+  const inAnimationConfig = animationConfig[properties.direction || 'top-to-bottom'].in;
```
Comment (AmreetKumarkhuntia) [inline src/lib/Toast/Toast.svelte]: nit : can pass default and use variable

**Review by AmreetKumarkhuntia — APPROVED** [review]

**Review by ARX1718 — APPROVED** [review]

File: src/lib/Toast/Toast.svelte [hunk 2]
```diff
@@ -1,11 +1,44 @@
...
+        };
+      default:
+        return {
+          in: { x: 0, y: -500, duration: 400 },
+          out: { x: 0, y: -100, duration: 800 }
+        };
+    }
+  }
+
+  const inAnimationConfig = getAnimationConfig(properties.direction).in;
```
Comment (swaroopvarma1) [inline src/lib/Toast/Toast.svelte]: no need of multiple variable initialization

File: src/lib/Toast/Toast.svelte [hunk 3]
```diff
@@ -1,11 +1,43 @@
 <script lang="ts">
   import { createEventDispatcher, onDestroy, onMount } from 'svelte';
   import { fly } from 'svelte/transition';
-  import { defaultToastProperties, type ToastProperties } from './properties';
+  import { defaultToastProperties, type ToastProperties, type ToastDirection } from './properties';
 
   export let properties: ToastProperties = defaultToastProperties;
   const dispatch = createEventDispatcher();
 
+  function getAnimationConfig(toastDirection: ToastDirection | undefined) {
```
Comment (swaroopvarma1) [inline src/lib/Toast/Toast.svelte]: null

File: src/lib/Toast/Toast.svelte [hunk 4]
```diff
@@ -1,11 +1,48 @@
 <script lang="ts">
   import { createEventDispatcher, onDestroy, onMount } from 'svelte';
   import { fly } from 'svelte/transition';
-  import { defaultToastProperties, type ToastProperties } from './properties';
+  import { defaultToastProperties, type ToastProperties, type ToastDirection } from './properties';
 
   export let properties: ToastProperties = defaultToastProperties;
   const dispatch = createEventDispatcher();
 
+  interface AnimationConfig {
```
Comment (swaroopvarma1) [inline src/lib/Toast/Toast.svelte]: type

**Review by swaroopvarma1 — APPROVED** [review]

**Review by sishir2001 — APPROVED** [review]

**Review by iamHrithikRaj — APPROVED** [review]

**Review by swaroopvarma1 — APPROVED** [review]

File: src/lib/types.ts [hunk 5]
```diff
@@ -36,3 +36,11 @@ export type TextTransformer = (text: string) => string;
  * @description Type Map for All possible output of an Input Filed Validation
  */
 export type ValidationState = 'Valid' | 'InProgress' | 'Invalid';
+
+/**
+ * @description Type for animation configuration
+ */
+export type FlyAnimationConfig = {
+  in: { x: number; y: number; duration: number };
```
Comment (sinha-sahil) [inline src/lib/types.ts]: This `{ x: number; y: number; duration: number }` can be a type and can be used with `in` & `out`.
Lets check if svelte transition exposes this type, if yes, lets use it.

File: src/lib/Toast/properties.ts [hunk 6]
```diff
@@ -1,11 +1,18 @@
...
+export type ToastDirection = 'left-to-right' | 'right-to-left' | 'top-to-bottom' | 'bottom-to-top';
 
 export type ToastProperties = {
   duration: number;
   leftIcon?: string;
   message: string;
   rightIcon?: string;
   type?: ToastType;
+  direction?: ToastDirection;
+  overContent?: boolean;
```
Comment (sinha-sahil) [inline src/lib/Toast/properties.ts]: `overlapPage` sounds more expressive.

File: src/lib/Toast/Toast.svelte [hunk 7]
```diff
@@ -66,11 +115,20 @@
...
     z-index: var(--toast-z-index, 1000);
     display: var(--taost-dispay, flex);
-    position: var(--toast-position, absolute);
+    top: var(--toast-top, 10px);
     left: var(--toast-left, 0);
     right: var(--toast-right, 0);
     background-color: var(--default-background-color, #87ceeb);
   }
+
+  .overContent {
```
Comment (sinha-sahil) [inline src/lib/Toast/Toast.svelte]: classes must be in `kebab-case`

File: src/lib/Toast/Toast.svelte [hunk 8]
```diff
@@ -31,9 +78,11 @@
 
 {#if showToast}
   <div
-    class="toast {properties.type ?? ''}"
-    in:fly={{ x: 500, y: 0, duration: 400 }}
-    out:fly={{ x: 500, y: 0, duration: 800 }}
+    class="toast {properties.type ?? ''} {properties?.overContent === false
+      ? 'insideContent'
```
Comment (sinha-sahil) [inline src/lib/Toast/Toast.svelte]: lets add separate class check for neatness.

`class:overlap-content={...check_for_should_overlap...}`

File: src/lib/Toast/Toast.svelte [hunk 9]
```diff
@@ -17,6 +22,48 @@
...
+  function getAnimationConfig(
+    toastDirection: ToastDirection | null = null,
+    overContent: boolean
+  ): FlyAnimationConfig {
+    let inX: number = 0;
+    let inY: number = 0;
+    let outX: number = 0;
+    let outY: number = 0;
+
+    switch (toastDirection) {
```
Comment (sinha-sahil) [inline src/lib/Toast/Toast.svelte:43]: Lets add comments on whats being done here.

**Review by swaroopvarma1 — APPROVED** [review]

File: src/lib/Toast/Toast.svelte [hunk 10]
```diff
@@ -32,8 +93,9 @@
 {#if showToast}
   <div
     class="toast {properties.type ?? ''}"
-    in:fly={{ x: 500, y: 0, duration: 400 }}
-    out:fly={{ x: 500, y: 0, duration: 800 }}
+    class:inside-content={!(properties?.overlapPage ?? true)}
```
Comment (sinha-sahil) [inline src/lib/Toast/Toast.svelte]: `no-page-overlap`

**Pull request merged**